packages/react-server/src/ReactFizzServer.js

```diff
@@ -411,11 +417,19 @@ function popComponentStackInDEV(task: Task): void {
   }
 }
 
-function logRecoverableError(request: Request, error: mixed): void {
+function logRecoverableError(request: Request, error: any): ?string {
   // If this callback errors, we intentionally let that error bubble up to become a fatal error
   // so that someone fixes the error reporting instead of hiding it.
   const onError = request.onError;
-  onError(error);
+  let errorMsg = onError(error);
+  if (__DEV__) {
```

Comment:
If you provide a hash in DEV should we include that too?

Reply:
Hmm, I don't think it'd be that useful? Surely the hash would be used to get this original error message anyways?